WordPress/src/main/java/org/wordpress/android/models/ReaderBlog.java

```diff
@@ -25,6 +26,28 @@
     private String mImageUrl;
     private String mFeedUrl;
 
+    public static ReaderBlog fromRecommendedBlogJson(@NotNull JSONObject json) {
```

Comment:
Moved this to a separate method to not increase the complexity of `ifs` statements more.

Reply:
These are also different

`feed_id` / `feed_ID`
`url` / `URL`
`blog_id` / "blog_ID" or "ID"
`feed_url` / `feed_URL`

Reply:
My reasoning: 
Usually, when I work with multiple backends I'm creating separate models that correspond to each API response body. The advantage is that if one API's response changes, changing just this one model is enough. 
Here I would say is a similar problem, but instead of creating a new model I've decided to add a new method with that parsing logic. If the API response changes I don't need to modify the if logic, I will just update this one method that is responsible for this.

Reply:
I see and your reasoning makes sense. At the same time I feel like if we have an inconsistent field names across the endpoints we might end up with tens of different parsing logics which are almost identical. It might get quite difficult to get oriented in and it would probably be highly error-prone. Since this is a new endpoint I'd consider pinging someone from the API and asking them if they considered making the endpoints consistent. Perhaps, there is a reason why the attribute names differ, but if there isn't I think introducing such inconsistency might bite us in the future. Wdyt?

Either case, I wouldn't block this PR on it. We can "refactor" the code if we decide to make the changes on the API.

Reply:
> Since this is a new endpoint I'd consider pinging someone from the API and asking them if they considered making the endpoints consistent.

I will do that. Tbh I assumed this is the new API and they don't want to be restricted by the old one? But is this true? 

 > We can "refactor" the code if we decide to make the changes on the API.

👍

Reply:
Hi folks - thanks for the feedback on this. On reflection I think it's probably best to keep the naming consistent with the v1 API, and include a separate `site` object inside each recommendation that mirrors the key naming of `/v1.1/read/site/:site`. I'll work on changing this.

Reply:
I've tackled this in D50244-code and put that up for review 👍 The response will look like this:

<img width="808" alt="Screen Shot 2020-09-28 at 14 52 21" src="https://user-images.githubusercontent.com/17325/94382759-65d54880-019a-11eb-8c33-2704c4236d9d.png">

The posts returned by the endpoint use the v1-style key naming, so it does feel right to be consistent. Thanks for raising it!

Just to provide some background on why I originally considered a different path here, my main historical gripes are:
* site and blog (and site_id/blog_id) are used interchangeably - generally the API refers to `site` but in the wpcom codebase it's usually `blog` and `site_id` means something else entirely.
* the key naming of v1 is a bit weird - mostly camel case, except for ID and URL (so we have prefer_feed but then also feed_ID and feed_URL) - we have to override our PHP linter to ignore these inconsistencies.
* we tend to include unnecessary site info in similar API responses that's never used by the client (for example, `/read/site/:site` returns four flags about which editor experience is enabled, which we never use), and I was aiming for the minimum in this case to keep the response as small as we can.

Reply:
Just a heads up that this change is going live today 🚢

Reply:
Thanks @bluefuton 💪 ! 
PR updated @ashiagr & @malinajirka